partcad/src/partcad/project_factory_git.py

```
@@ -86,80 +137,114 @@
+                                pc_logging.debug("Refreshing the GIT branch: %s" % short_branch_name)
+                                origin.pull(short_branch_name)
+                                self.ctx.stats_git_ops += 1
+                                os.utime(guard_path, (now, now))
+                        else:
+                            # Import a specific revision
+                            if user_config.force_update:
+                                # Ensure "before" doesn't match the desired revision
+                                before = ""
+                            else:
+                                # Read the revision name from the guard file
+                                with open(guard_path, "r") as f:
```

Ruff (0.8.2)

Unnecessary open mode parameters

Remove open mode parameters

(UP015)